### PR TITLE
Objdetect: QRCode detector - Mat assertion, added binding tests

### PR DESCRIPTION
**Pull request by allnes**
Issue: [https://github.com/opencv/opencv/issues/14709](https://github.com/opencv/opencv/issues/14709)

### CODE REVIEW

File: modules/objdetect/src/qrcode.cpp [hunk 1]
```diff
@@ -1084,7 +1084,7 @@ cv::String QRCodeDetector::decode(InputArray in, InputArray points,
         inarr = gray;
     }
 
-    CV_Assert(points.isVector());
+    CV_Assert(points.isVector() || points.isMat());
     vector<Point2f> src_points;
     points.copyTo(src_points);
```
Comment (alalek) [inline modules/objdetect/src/qrcode.cpp]: Due `.copyTo()` processing this assertion is useless.

File: modules/python/test/test_qrcode_detect.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python
+'''
+===============================================================================
+QR code detect and decode pipeline.
```
Comment (alalek) [inline modules/python/test/test_qrcode_detect.py]: Move test to the corresponding module:
https://github.com/opencv/opencv/tree/3.4/modules/objdetect/misc/python/test

File: modules/objdetect/misc/java/test/QRCodeDetectorTest.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package org.opencv.test.objdetect;
+
+import org.opencv.core.Mat;
+import org.opencv.objdetect.QRCodeDetector;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.opencv.objdetect.Objdetect;
```
Comment (alalek) [inline modules/objdetect/misc/java/test/QRCodeDetectorTest.java]: Used this import?

File: modules/objdetect/misc/java/test/QRCodeDetectorTest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import org.opencv.test.OpenCVTestCase;
+import org.opencv.test.OpenCVTestRunner;
```
Comment (alalek) [inline modules/objdetect/misc/java/test/QRCodeDetectorTest.java]: used?

File: modules/objdetect/misc/java/test/QRCodeDetectorTest.java (hunk 3 above)
Comment (allnes) [reply modules/objdetect/misc/java/test/QRCodeDetectorTest.java]: @alalek done

File: modules/objdetect/misc/java/test/QRCodeDetectorTest.java (hunk 4 above)
Comment (allnes) [reply modules/objdetect/misc/java/test/QRCodeDetectorTest.java]: @alalek removed

File: modules/python/test/test_qrcode_detect.py (hunk 2 above)
Comment (allnes) [reply modules/python/test/test_qrcode_detect.py]: @alalek replaced

**Pull request merged**